### PR TITLE
Fix rotating member bugs

### LINKED ISSUE #2441: DOSFCANL_ENKF branch of scripts/exgdas_enkf_sfc.sh broken
### What is wrong?

C96C48_ufs_hybatmDA CI job enkfgdasesfc fails with
```
+ preamble.sh[143]: set +x
generate_com :: COM_ATMOS_RESTART_MEM=/scratch1/NCEPDEV/stmp2/Russ.Treadon/COMROOT/prjedi/enkfgdas.20240224/00/mem001/model_data/atmo\
s/restart
+ exgdas_enkf_sfc.sh[196]: RUN=enkfgdas
/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/test/scripts/exgdas_enkf_sfc.sh: line 196: gmemchar: unbound variable
++ exgdas_enkf_sfc.sh[1]: postamble exgdas_enkf_sfc.sh 1711646776 1
++ preamble.sh[70]: set +x
End exgdas_enkf_sfc.sh at 17:26:18 with error code 1 (time elapsed: 00:00:02)
```

### What should have happened?

C96C48_ufs_hybatmDA CI job enkfgdasesfc should run to completion with no error

### What machines are impacted?

All or N/A

### Steps to reproduce

1. clone and install g-w `develop`
2. enable C96C48_ufs_hybatmDA CI
3. rocotorun up to 20240224 00Z enkfgdasesfc
4. rocotoboot 20240224 00Z enkfgdasesfc.  Job will fail.

### Additional information

IAU is not yet functional in JEDI ATM DA.  This is why `ci/cases/yamls/ufs_hybatmDA_defaults.ci.yaml` includes
```
base:
  DOIAU: "NO"
```

### Do you have a proposed solution?

g-w commit [47302153](https://github.com/NOAA-EMC/global-workflow/commit/47302153f13f6b23539be841b78ed78664599c08) added the following to the `[ $DOIAU = "YES" ]` section of `exgdas_enkf_sfc.sh`
```
            smem=$((imem + mem_offset))
            if (( smem > 80 )); then
               smem=$((smem - 80))
            fi
            gmemchar="mem"$(printf %03i "$smem")
```
The same addition is needed in the `[ $DOSFCANL_ENKF = "YES" ]` section of the script.

### PR DESCRIPTION
**Pull request by TravisElless-NOAA**
# Description

When PR #2427 introduced the rotating subset of member guess states for the early-cycle EnKF, the rotating member calculation function was omitted from the DOSFCANL_ENKF if block in the enkf surface script. This PR adds this feature to this section. This PR also removes hard coded values that were included in this function and are replaced with a variable equal to the number of late cycle members.   

  Resolves #2441 

# Type of change

- Bug fix (fixes something broken)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
-Cycled test Orion/WCOSS2

# Checklist
- [ ] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Comment by TravisElless-NOAA** [comment]
Sorry for missing this the first time around @RussTreadon-NOAA

**Review by RussTreadon-NOAA — APPROVED** [review]
Looks good to me.  This is the change I made in my local copy of `scripts/exgdas_enkf_sfc.sh`.  It works!

File: scripts/exgdas_enkf_sfc.sh [hunk 1]
```diff
@@ -186,7 +186,11 @@ if [ $DOSFCANL_ENKF = "YES" ]; then
         export TILE_NUM=$n
 
         for imem in $(seq 1 $NMEM_ENS); do
-
+            smem=$((imem + mem_offset))
+            if (( smem > 80 )); then
+               smem=$((smem - 80))
```
Comment (aerorahul) [inline scripts/exgdas_enkf_sfc.sh]: Should `80` be replaced w/ `NMEM_ENS`

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved.

**Pull request merged**